gui/main.js

```diff
@@ -518,57 +518,9 @@ const execSync = (cmd) => {
-  const syncRootId = windowsSyncRootId(windowsSID, 'default')
-  const clsid = uuid.v4().toUpperCase()
-  const exePath = process.argv[0]
-
-  execSync(`reg add HKLM\\Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\SyncRootManager\\${syncRootId} /v DisplayNameResource /t REG_SZ /d "Cozy Drive" /f`)
-  execSync(`reg add HKLM\\Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\SyncRootManager\\${syncRootId} /v IconResource /t REG_SZ /d "${exePath}" /f`)
-  execSync(`reg add HKLM\\Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\SyncRootManager\\${syncRootId}\\UserSyncRoots /v ${windowsSID} /t REG_SZ /d "${syncPath}" /f`)
-
-  execSync(`reg add HKCU\\Software\\Classes\\CLSID\\{${clsid}} /ve /t REG_SZ /d "Cozy Drive" /f`)
-  execSync(`reg add HKCU\\Software\\Classes\\CLSID\\{${clsid}}\\DefaultIcon /ve /t REG_SZ /d "${exePath}" /f`)
-  execSync(`reg add HKCU\\Software\\Classes\\CLSID\\{${clsid}} /v System.IsPinnedToNameSpaceTree /t REG_DWORD /d 0x1 /f`)
-  execSync(`reg add HKCU\\Software\\Classes\\CLSID\\{${clsid}} /v SortOrderIndex /t REG_DWORD /d 0x42 /f`)
-  execSync(`reg add HKCU\\Software\\Classes\\CLSID\\{${clsid}}\\InProcServer32 /ve /t REG_EXPAND_SZ /d %%systemroot%%\\system32\\shell32.dll /f`)
-  execSync(`reg add HKCU\\Software\\Classes\\CLSID\\{${clsid}}\\Instance /v CLSID /t REG_SZ /d {0E5AAE11-A475-4c5b-AB00-C66DE400274E} /f`)
-  execSync(`reg add HKCU\\Software\\Classes\\CLSID\\{${clsid}}\\Instance\\InitPropertyBag /v Attributes /t REG_DWORD /d 0x11 /f`)
-  execSync(`reg add HKCU\\Software\\Classes\\CLSID\\{${clsid}}\\Instance\\InitPropertyBag /v TargetFolderPath /t REG_EXPAND_SZ /d "${syncPath}" /f`)
-  execSync(`reg add HKCU\\Software\\Classes\\CLSID\\{${clsid}}\\ShellFolder /v FolderValueFlags /t REG_DWORD /d 0x28 /f`)
-  execSync(`reg add HKCU\\Software\\Classes\\CLSID\\{${clsid}}\\ShellFolder /v Attributes /t REG_DWORD /d 0xF080004D /f`)
-  execSync(`reg add HKCU\\Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\Desktop\\NameSpace\\{${clsid}} /ve /t REG_SZ /d "Cozy Drive" /f`)
-  execSync(`reg add HKCU\\Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\HideDesktopIcons\\NewStartPanel /v {${clsid}} /t REG_DWORD /d 0x1 /f`)
-
-  return {syncRootId, clsid}
-}
-
-const windowsUnregisterCloudStorageProvider = (windowsConfig) => {
-  const {syncRootId, clsid} = windowsConfig
-
-  execSync(`reg delete HKLM\\Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\SyncRootManager\\${syncRootId} /f`)
-  execSync(`reg delete HKCU\\Software\\Classes\\CLSID\\{${clsid}} /f`)
-  execSync(`reg delete HKCU\\Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\Desktop\\NameSpace\\{${clsid}} /f`)
-  execSync(`reg delete HKCU\\Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\HideDesktopIcons\\NewStartPanel /v {${clsid}} /f`)
-}
-*/
+const win10PinToHome = (path) => {
+  execSync(`powershell -Command "(New-Object -COM shell.application).Namespace('${path}').Self.InvokeVerb('pintohome')"`)
```

Comment:
More a user concern than a security issue, a legitimate path could contain a single quote.

It would be nice to progressively extract some functions from `gui/main.js` and unit-test them... WDYT?